### PR TITLE
[KYUUBI #5652][AUTHZ] CreateTable related command support path privilege

### LINKED ISSUE #5652: [Subtask] Support path privilege of CreateTable
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Support
CreateTable
AlterTableAddPartitionCommand
AlterTableSetLocationCommand

### Parent issue

#5368

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
To close #5652 
CreateTable related command support path privilege

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @yaooqinn

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -1240,4 +1240,70 @@ class HiveCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
...
+                     |CREATE TABLE $db1.$table2
+                     |LOCATION '$path'
+                     |AS
+                     |SELECT * FROM $db1.$table1
+                     |""".stripMargin)))(
+              s"does not have [select] privilege on [$db1/$table1/id,$db1/$table1/scope], " +
+                s"[create] privilege on [$db1/$table2/id,$db1/$table2/scope], " +
+                s"[write] privilege on [[file://$path, file://$path/]]")
+          }
+        }
```
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:1305]:
```suggestion
          interceptContains[AccessControlException](
            doAs(
              someone,
              sql(
                s"""
                   |CREATE TABLE $db1.$table2
                   |LOCATION '$path'
                   |AS
                   |SELECT * FROM $db1.$table1
                   |""".stripMargin))) {
            if (isSparkV35OrGreater) {
              s"does not have [select] privilege on [$db1/$table1/id,$db1/$table1/scope], " +
                s"[create] privilege on [$db1/$table2/id,$db1/$table2/scope], " +
                s"[write] privilege on [[file://$path, file://$path/]]"
            } else {
              s"does not have [select] privilege on [$db1/$table1/id,$db1/$table1/scope], " +
                s"[create] privilege on [$db1/$table2/id,$db1/$table2/scope], " +
                s"[write] privilege on [[$path, $path/]]"
            }
          }
```

**Comment by AngersZhuuuu** [comment]
Conflict resolved

**Review by bowenliang123 — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**